src/Compilers/CSharp/Portable/Symbols/TypeMap.cs

```diff
@@ -20,9 +20,7 @@ internal sealed class TypeMap : AbstractTypeParameterMap
 
         internal static ImmutableArray<TypeSymbolWithAnnotations> TypeParametersAsTypeSymbolsWithAnnotations(ImmutableArray<TypeParameterSymbol> typeParameters)
         {
-            return typeParameters.SelectAsArray((tp) =>
-                TypeSymbolWithAnnotations.Create(NonNullTypesNullContext.Instance, tp, isAnnotated: false, customModifiers: ImmutableArray<CustomModifier>.Empty)
-                );
+            return typeParameters.SelectAsArray((tp) => TypeSymbolWithAnnotations.Create(tp));
```

Comment:
There seem to be a few places using this method of creation. We could consider extracting this into a helper and doing the iteration by hand, saving the lambda allocations.

Reply:
A helper method should continue to use a lambda since the additional memory for a single static lambda instance is probably similar to the extra instructions from a loop.